protocols/rendezvous/src/client.rs

```diff
@@ -82,13 +79,15 @@ impl Behaviour {
     ///
     /// External addresses are either manually added via [`libp2p_swarm::Swarm::add_external_address`] or reported
     /// by other [`NetworkBehaviour`]s via [`ToSwarm::ExternalAddrConfirmed`].
-    pub fn register(&mut self, namespace: Namespace, rendezvous_node: PeerId, ttl: Option<Ttl>) {
+    pub fn register(
+        &mut self,
+        namespace: Namespace,
+        rendezvous_node: PeerId,
+        ttl: Option<Ttl>,
+    ) -> Result<(), RegisterError> {
```

Comment:
So here is another idea: This error has three variants but only two are returned from this function.

My suggestion would be to remove the `Remote` variant from `RegisterError` and instead put the information from `Remote` directly into the variant from the `Event` enum that currently contains the `RegisterError` variant.

That way, a user is not faced with "impossible" errors.

Reply:
Marking this as unresolved because it is not yet addressed.